interactors/nonceHandlerV2/addressNonceHandler_test.go

```diff
@@ -96,6 +96,22 @@ func TestAddressNonceHandler_ApplyNonceAndGasPrice(t *testing.T) {
 		err = anh.ApplyNonceAndGasPrice(context.Background(), &tx)
 		require.Nil(t, err)
 	})
+	t.Run("same transaction but with different nonce should work", func(t *testing.T) {
+		t.Parallel()
+
+		tx1 := createDefaultTx()
+		tx2 := createDefaultTx()
+		tx2.Nonce++
+
+		anh, err := NewAddressNonceHandlerWithPrivateAccess(&testsCommon.ProxyStub{}, testAddress)
+		require.Nil(t, err)
+
+		_, err = anh.SendTransaction(context.Background(), &tx1)
```

Comment:
Isn't the actual use case the one where we do smth like this?
- ApplyNonce(tx1)
- Send(tx1)
- ApplyNonce(tx2)
- Send(tx2)

Reply:
Changed. Anyway, the problem occurs only when we try to apply the nonce for the new (same) transaction after we sent the first one